tests/core/gadget-config-defaults-vitality/task.yaml

```diff
@@ -0,0 +1,155 @@
+summary: |
+   Test that vitality defaults specified in the gadget are picked up
+   for first boot snaps
+
+# the test is only meaningful on core devices
+# TODO:UC20: enable for UC20, it assumes /var/lib/snapd/seed/assertions/model
+#            which we don't have currently
+systems: [ubuntu-core-1*]
+
+environment:
+    GADGET_FILE/vitality_hint: gadget-vitality-hint.yaml
+
+prepare: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    #shellcheck source=tests/lib/systemd.sh
+    . "$TESTSLIB"/systemd.sh
+    #shellcheck source=tests/lib/systems.sh
+    . "$TESTSLIB"/systems.sh
+
+    SUFFIX=
+    MODEL=developer1-pc-w-config.model
+    if is_core18_system; then
+        SUFFIX=-core18
+        MODEL=developer1-pc-18-w-config.model
+    fi
+
+    systemctl stop snapd.service snapd.socket
```

Comment:
Could we use a helper like snap-mgmt? I would feel better if we had one place that "removes state" that needs patching, instead of many, implemented by various tests.

Reply:
we have a bunch of tests that do this kind of things but details are never quite exactly the same, we could factor it out somehow but it unclear it would be more readable than just doing it, especially because of the details being not quite the same aspect